pulp_2to3_migrate/app/serializers.py

```diff
@@ -29,12 +35,44 @@ def validate(self, data):
         """
         Validate that the Serializer contains valid data.
 
-        TODO:
-        Validate JSON structure of migration_plan.
-        Check validity of the JSON content:
-         - migration for requested plugins is supported
-
+        Validates JSON structure of migration_plan.
+        Checks pulp2 and pulp3 plugins are installed.
         """
+        schema = json.loads(SCHEMA)
+        validator = Draft7Validator(schema)
+        loaded_plan = json.loads(data['plan'])
+        err = []
+        for error in sorted(validator.iter_errors(loaded_plan), key=str):
+            err.append(error.message)
+        if err:
+            raise serializers.ValidationError(
+                    _("Provided Migration Plan format is invalid:'{}'".format(err))
+            )
+        plugins_to_migrate = set()
+        for plugin_type in loaded_plan['plugins']:
+            plugins_to_migrate.add(plugin_type['type'])
+        if len(loaded_plan['plugins']) != len(plugins_to_migrate):
+            raise serializers.ValidationError(
+                _("Provided Migration Plan contains same plugin type specified more that once.")
+            )
+        # MongoDB connection initialization
+        connection.initialize()
+        db = connection.get_database()
+        for plugin in plugins_to_migrate:
```

Comment:
This probably won't work for checking installed plugins in Pulp 3 because of potentially different names in Pulp2 and Pulp3.
E.g. "iso" in Pulp 2 is the "file" in Pulp 3.
We probably need to intorduce a new constant/mapping if we can't get this information form existing constants in the migration plugin.